src/app-layout/split-panel/split-panel-utils.ts

```diff
@@ -10,17 +12,23 @@ import { SPLIT_PANEL_MIN_WIDTH, SPLIT_PANEL_SCROLLBAR_MARGIN } from './constants
  * minimum width exceeds this value then there is not enough horizontal space and we must
  * force it to the bottom position.
  */
-export function checkSplitPanelForcedPosition({
+export function useSplitPanelForcedPosition({
   isMobile,
   splitPanelMaxWidth,
 }: {
   isMobile: boolean;
   splitPanelMaxWidth: number;
 }) {
+  const isForcedRef = useRef<boolean>(false);
```

Comment:
Would it be easier to use closures instead of turning this function into react hook?

Reply:
It was not difficult, I only had to rename a couple files :). I felt like doing that and resort to refs would be a more conventional and therefore less risky option than introducing my own solutions.